### PR TITLE
jonna's todo

### PR DESCRIPTION
**Pull request by jonnaru**
**Link to my deployed project:**
https://todo-list-by-jonna.netlify.app/

My reflections on how this week's project turned out:
much to learn

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,55 @@
...
+import { TaskList } from "./components/TaskList";
+import { AddTask } from "./components/AddTask";
+import { HandleTasks } from "./components/HandleTasks";
+import { Header } from "./components/Header";
+
+import styled from "styled-components";
+
+const Wrapper = styled.section`
+  padding: 10px;
+  background: #d3d3d3;
```
Comment (johannarexin) [inline code/src/App.js:15]: Nice with styled components!

File: code/src/components/HandleTasks.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { todos } from "reducers/todos";
+
+import styled from "styled-components";
+import { Button } from "lib/Buttons";
```
Comment (johannarexin) [inline code/src/components/HandleTasks.js:6]: Clean import, good! I think Maks said in one lecture that the imports should be structured like this: 
1. Imports from "outside" like React, useSelector, useDispatch and so on.
2. Imports from our own code like { todos }, components etcetera.
3. Imports with styling.

Just good to have in mind in the future :)

File: code/src/components/HandleTasks.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          Complete all
+        </Button>
+        <Button
+          type="button"
+          onClick={() => {
+            dispatch(todos.actions.removeAllTasks());
+          }}
+        >
+          Remove all
+        </Button>
```
Comment (johannarexin) [inline code/src/components/HandleTasks.js:49]: Nice with two buttons and the opportunities to either remove all tasks or complete all tasks.

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { todos } from "reducers/todos";
+import { useDispatch, useSelector } from "react-redux";
+
+import swal from "sweetalert";
+
+import styled from "styled-components";
+import { Button, CircleButton } from "lib/Buttons";
```
Comment (johannarexin) [inline code/src/components/Header.js:8]: Same as my previous comment about the order of imports :)

File: code/src/lib/Checkbox.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    top: 5px;
+    left: 4px;
+    height: 17px;
+    transition: 0.1s;
+    width: 17px;
+    background-image: url(../assets/check.svg);
+    background-repeat: no-repeat;
+    background-size: cover;
+    opacity: 0;
+  }
```
Comment (johannarexin) [inline code/src/lib/Checkbox.js:28]: Nice! Didn't know you could write like this, learned something new :) Thanks!

File: code/src/components/HandleTasks.js (hunk 2 above)
Comment (jonnaru) [reply code/src/components/HandleTasks.js:6]: Thank you, Johanna!  I'll change this :)))

File: code/src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        },
+      },
+    }).then((value) => {
+      if (value) {
+        const name =
+          value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
+        dispatch(todos.actions.setUserName(name));
+      }
+    });
+  };
```
Comment (HIPPIEKICK) [inline code/src/components/Header.js:40]: Such a nice feature 🤩

File: code/src/reducers/todos.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    ],
+  },
+  reducers: {
+    setUserName: (state, action) => {
+      state.username = action.payload;
+    },
+    addTask: (state, action) => {
+      state.tasks.push(action.payload);
+    },
+    doneTask: (state, action) => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/todos.js:20]: Maybe `toggleDoneTask`? :)

**Pull request closed without merging**